test/unit/dao/FeiDao.test.ts

```diff
@@ -0,0 +1,189 @@
+import { expectRevert, time, getCore, getAddresses } from '../../helpers';
+import { expect } from 'chai';
+import hre, { artifacts, ethers, network } from 'hardhat';
+import { Signer } from 'ethers';
+import { TransactionResponse } from '@ethersproject/providers';
+
+const FeiDAO = artifacts.readArtifactSync('FeiDAO');
+const Timelock = artifacts.readArtifactSync('Timelock');
+const Tribe = artifacts.readArtifactSync('Tribe');
+
+const toBN = ethers.BigNumber.from;
+
+describe('FeiDAO', function () {
+  let userAddress: string;
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [addresses.userAddress];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress } = await getAddresses());
+
+    await network.provider.request({
+      method: 'hardhat_reset',
+      params: []
+    });
+
+    this.core = await getCore();
+    const tribeAddress = await this.core.tribe();
+
+    // Deploy timelock with vanilla user admin
+    const timelockDeployer = await ethers.getContractFactory(Timelock.abi, Timelock.bytecode);
+    this.timelock = await timelockDeployer.deploy(userAddress, 1, 1); // delay and min delay 1s
```

Comment:
Since you already wrote this I hate to ask, but can you remove all references to `this` given the importance of the upgrade?

Reply:
Fixed. I didn't type Core because it had some upstream stuff I didn't want to mess with